### PR TITLE
Refactor/cog for covering optimization

### PR DESCRIPTION
**Pull request by jacott**
### Changes

Prepare for optimizing cog creation and covering

#### Source Code Documentation Tasks:
- [ ] README updated (where applicable)
- [ ] CHANGELOG (Unreleased section) updated
- [ ] Docstrings / comments included to help explain code

#### User Documentation Tasks:
- [ ] Confluence user guide updated (where applicable)

#### Testing Tasks:
- [ ] Added tests that fail without this change
- [ ] All tests are passing in development environment
- [ ] Reviewers assigned
- [ ] Linked to main issue for ZenHub board

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "author": "",
   "license": "MIT",
   "scripts": {
-    "lint": "eslint 'packages/*/src/**/*.ts' --quiet --fix",
+    "lint": "eslint 'packages/*/src/**/*.ts' --quiet --fix --report-unused-disable-directives",
```
Comment (blacha) [inline package.json:9]: Put this in the eslintrc.js

Comment (blacha) [reply package.json:9]: https://eslint.org/docs/user-guide/configuring#report-unused-eslint-disable-comments

File: packages/cog/src/cog/quadkey.vrt.ts [hunk 2]
```diff
@@ -209,4 +206,21 @@ export const Cutline = {
...
         return target;
     },
+
+    /**
+     * Load a geojson cutline from the file-system and convert to one multi-polygon with any holes removed
+     *
+     * @param path the path of the cutline to load. Can be `s3://` or local file path.
+     */
+    async loadCutline(path: string): Promise<FeatureCollection> {
+        const inCutline = JSON.parse((await FileOperator.create(path).read(path)).toString());
```
Comment (blacha) [inline packages/cog/src/cog/quadkey.vrt.ts]: We do this a lot, how about a 

`FileOperator.readJson`

File: packages/geo/src/quad.key.ts [hunk 3]
```diff
@@ -38,6 +39,27 @@ export const QuadKey = {
...
+    toBbox(quadKey: string): [number, number, number, number] {
+        return tileToBBOX(quadkeyToTile(quadKey));
+    },
+
+    /**
+     * Convert a point to a quadKey at a given zoom level
+     * @param point at lat, lon pair
+     * @param zoom the number of digits in the resulting quadkey
+     */
+    pointToQuadKey([lat, lon]: number[], zoom: number): string {
```
Comment (blacha) [inline packages/geo/src/quad.key.ts:51]: `[x, y]`

`[lon, lat]`

File: packages/cog/src/cog/__test__/job.cutline.test.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import { FileOperatorSimple } from '@basemaps/lambda-shared';
+import { MultiPolygon } from 'geojson';
+import { JobCutline } from '../job.cutline';
+import * as o from 'ospec';
+import 'source-map-support/register';
```
Comment (blacha) [inline packages/cog/src/cog/__test__/job.cutline.test.ts:5]: remove

File: packages/cog/src/cog/builder.ts [hunk 5]
```diff
@@ -6,51 +6,29 @@ import { createHash } from 'crypto';
...
     q: Limit;
-    cover: TileCover;
-    maxTileCount: number;
-    maxTileZoom: number;
 
     /**
      * @param concurrency number of requests to run at a time
      */
-    constructor(concurrency: number, tileCountMax = 25, tileZoomMax = 13) {
+    constructor(concurrency: number, public maxTileCount = 25, public maxTileZoom = 13, public logger: LogType) {
```
Comment (blacha) [inline packages/cog/src/cog/builder.ts:23]: I dont think we use this sytanx anywhere in the codebase.

File: packages/cog/src/cog/tile.cover.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import { GeoJson, QuadKey } from '@basemaps/geo';
+import * as MapBoxCover from '@mapbox/tile-cover';
+
+export const TileCover = {
```
Comment (blacha) [inline packages/cog/src/cog/tile.cover.ts:4]: Could this live in `@basemaps/geo`  ?

File: package.json (hunk 1 above)
Comment (jacott) [reply package.json:9]: This only add to warnings; not failure. Keeping command line option

**Pull request merged**